.github/workflows/release-versoview.yml

```diff
@@ -0,0 +1,126 @@
+          nix-channel --update nixpkgs
+
+      - name: Build
+        run: nix-shell shell.nix --run 'cargo build --release'
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: verso-x86_64-unknown-linux-gnu
+          path: ./target/release/verso
+
+  build-others:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+            exe: .exe
+          # - os: windows-latest
+          #   target: aarch64-pc-windows-msvc
+          #   exe: .exe
+          - os: macos-14
+            target: aarch64-apple-darwin
+          # - os: macos-13
+          #   target: x86_64-apple-darwin
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.4
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install mako
+        run: python -m pip install mako
+
+      - name: Install scroop
+        if: ${{ startsWith(matrix.os, 'windows') }}
+        uses: MinoruSekine/setup-scoop@v3
+      - name: Install windows dependencies
+        if: ${{ startsWith(matrix.os, 'windows') }}
+        run: |
+          scoop install llvm cmake
+
+      - name: Install mac dependencies
+        if: ${{ startsWith(matrix.os, 'macos') }}
+        run: |
+          brew install cmake
+          curl https://gstreamer.freedesktop.org/data/pkg/osx/1.24.6/gstreamer-1.0-1.24.6-universal.pkg -o runtime.pkg
+          sudo installer -pkg runtime.pkg -target /
+          curl https://gstreamer.freedesktop.org/data/pkg/osx/1.24.6/gstreamer-1.0-devel-1.24.6-universal.pkg -o develop.pkg
+          sudo installer -pkg develop.pkg -target /
+
+      - name: Add rust target
+        run: rustup target add ${{ matrix.target }}
+
+      - name: Build
+        env:
+          SCCACHE_GHA_ENABLED: true
+          RUSTC_WRAPPER: "sccache"
+        run: cargo build --release --target ${{ matrix.target }}
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: verso-${{ matrix.target }}
+          path: ./target/${{ matrix.target }}/release/verso${{ matrix.exe }}
+
+  publish-release:
+    runs-on: ubuntu-latest
+    needs: [build-linux, build-others]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          path: artifacts
+
+      - working-directory: ./artifacts/verso-x86_64-pc-windows-msvc
+        run: zip -r ../verso-x86_64-pc-windows-msvc.zip *
+      - working-directory: ./artifacts/verso-aarch64-apple-darwin
+        run: zip -r ../verso-aarch64-apple-darwin.zip *
+      - working-directory: ./artifacts/verso-x86_64-unknown-linux-gnu
+        run: zip -r ../verso-x86_64-unknown-linux-gnu.zip *
```

Comment:
The compression should happen during uploading artifact I think.
And could we use tar.gz format?

Reply:
Will do